conda_smithy/templates/run_docker_build.tmpl

```diff
@@ -26,9 +26,12 @@ show_channel_urls: true
 CONDARC
 )
 
+HOST_USER_ID=$( hash docker-machine 2>/dev/null && docker-machine ssh $(docker-machine active) 'id -u $(whoami)' || id -u $(whoami) )
```

Comment:
> To be absolutely clear, I was talking about the user inside the container

So in this case you could just set HOST_USER_ID to 0, this would have the same effect as running docker as root.

> I have another question that will highlight my ignorance, but I'd like to know whether starting the container as root (as we currently do) and simply su to a different user has been considered. Conceptually it feels considerably simpler, are there are drawbacks that would stop us going down that root?

One of the problems is that the regular user needs write permission to mounted volumes, That is very tricky with docker and that is why a user is created on-the-fly in the entrypoint and owners for some directories are changed. If this is not done in the entrypoint, the user would have to do it manually. So actually it would not be that simpler.

I think the greater question is: what should be the default behaviour? In my opinion using root by default can bring more problems than benefits. Many commands change the default behaviour when running as root (see https://github.com/conda-forge/staged-recipes/pull/1468) or even refuse to run. So I think this is a good reason to run the container with a regular user by default, to avoid mistakes.